pkg/splunk/enterprise/configuration.go

```diff
@@ -556,20 +583,28 @@ func updateSplunkPodTemplateWithConfig(podTemplateSpec *corev1.PodTemplateSpec,
 		}
 	}
 
+	// Explicitly set the default value here so we can compare for changes correctly with current statefulset.
+	secretVolDefaultMode := int32(corev1.SecretVolumeSourceDefaultMode)
+
 	// add defaults secrets to all splunk containers
 	addSplunkVolumeToTemplate(podTemplateSpec, "secrets", corev1.VolumeSource{
 		Secret: &corev1.SecretVolumeSource{
-			SecretName: GetSplunkSecretsName(cr.GetIdentifier(), instanceType),
+			SecretName:  GetSplunkSecretsName(cr.GetIdentifier(), instanceType),
+			DefaultMode: &secretVolDefaultMode,
 		},
 	})
 
+	// Explicitly set the default value here so we can compare for changes correctly with current statefulset.
+	configMapVolDefaultMode := int32(corev1.ConfigMapVolumeSourceDefaultMode)
+
 	// add inline defaults to all splunk containers
 	if spec.Defaults != "" {
 		addSplunkVolumeToTemplate(podTemplateSpec, "defaults", corev1.VolumeSource{
 			ConfigMap: &corev1.ConfigMapVolumeSource{
 				LocalObjectReference: corev1.LocalObjectReference{
 					Name: GetSplunkDefaultsName(cr.GetIdentifier(), instanceType),
 				},
+				DefaultMode: &configMapVolDefaultMode,
```

Comment:
Since the default mode is already set above (validateVolumes) can't it be retrieved from the CR spec here instead of initializing with corev1.ConfigMapVolumeSourceDefaultMode.

Reply:
Comment above